hal/src/rtl872x/ble_hal.cpp

```diff
@@ -812,6 +1022,30 @@ int BleGap::init() {
         CHECK(setDeviceAddress(&addr_));
     }
 
+    uint8_t pairable = GAP_PAIRING_MODE_PAIRABLE;
+    CHECK_RTL(gap_set_param(GAP_PARAM_BOND_PAIRING_MODE, sizeof(pairable), &pairable));
+    // Bit mask: GAP_AUTHEN_BIT_BONDING_FLAG, GAP_AUTHEN_BIT_MITM_FLAG, GAP_AUTHEN_BIT_SC_FLAG
+    uint16_t authFlags = 0;
+    if (pairingConfig_.algorithm == BLE_PAIRING_ALGORITHM_AUTO) {
+        // Prefer LESC
+        authFlags = GAP_AUTHEN_BIT_SC_FLAG;
+    } else if (pairingConfig_.algorithm == BLE_PAIRING_ALGORITHM_LESC_ONLY) {
+        authFlags = GAP_AUTHEN_BIT_SC_ONLY_FLAG;
```

Comment:
Needs to be `GAP_AUTHEN_BIT_SC_ONLY_FLAG | GAP_AUTHEN_BIT_FLAG`. I've tested in both roles, works fine.

Reply:
Fixed.